master/static/srv/get_model_version_labels.sql

```diff
@@ -0,0 +1,11 @@
+WITH sorted_labels AS (
+  SELECT label
+  FROM (
+    SELECT id, UNNEST(labels) AS label
+    FROM model_versions
```

Comment:
Oof, so it seems we don't have a table for tags and we need to scan the whole table of taggable objects to get the list of them? Do we have an index on labels? If not we could use an index to precompute this list on write so we don't have to scan the whole table on read.

Reply:
we dont have a table for tags or index for labels. There are 2 types of index for array type. I'll check them out

Reply:
It's been a while since I've done this kind of thing but in addition to postgres specific stuff you might look into loose index scanning to just get the unique values from an index instead of all the values.

Reply:
related:

https://github.com/determined-ai/determined/blob/master/master/static/srv/get_experiment_labels.sql

btw [...] has 2 million experiments.

we are going with a different data model for tags as a part of trials comparison, due to this and race condition issues with the current approach.

basically just using a `jsonb` dict where keys are present if the tag exists for the object.

using GIN `jsonb_path_ops` index allows for efficient lookup based on keys.

I think if you did an agg function just using the [index compatible operators](https://www.postgresql.org/docs/current/gin-builtin-opclasses.html), it would be able to use the index for getting distinct tags as well. Have not validated yet though.

Reply:
the plan (based on team discussions) was to try the new data model/approach with trials, and then convert everywhere else we use tags after validating.

Reply:
i updated [this ticket](https://determinedai.atlassian.net/browse/DET-8185) to reflect the overall context

Reply:
I think this ticket is blocked on that more efficient tag representation. Let's table DET-8133 for now and revisit once we can fix the tag lookup.

Reply:
I created DET-8243 to track the issue of tag representation and I'm gonna pull tag typeahed out of the sprint for the time being.